Dockerfile

```diff
@@ -1,18 +1,82 @@
-FROM continuumio/miniconda3
+FROM nvidia/cuda:11.2.0-cudnn8-devel-ubuntu20.04 AS build
 
-WORKDIR /app
+# RNCAN certificate; uncomment (with right .cer name) if you are building behind a FW
+# COPY NRCan-RootCA.cer /usr/local/share/ca-certificates/cert.crt
+# RUN chmod 644 /usr/local/share/ca-certificates/cert.crt && update-ca-certificates
 
-# Create the environment:
-COPY environment.yml .
-RUN conda env create -f environment.yml
+ARG PYTHON_VERSION=3.9
+ARG CONDA_PYTHON_VERSION=3
+ARG CONDA_DIR=/opt/conda
 
-# Make RUN commands use the new environment:
-SHELL ["conda", "run", "-n", "geo_deep_env", "/bin/bash", "-c"]
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends git wget unzip bzip2 build-essential sudo \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
 
-# Make sure the environment is activated:
-RUN echo "Make sure flask is installed:"
-RUN python -c "import flask"
+# Install miniconda
+ENV PATH $CONDA_DIR/bin:$PATH
+RUN wget --quiet https://repo.continuum.io/miniconda/Miniconda$CONDA_PYTHON_VERSION-latest-Linux-x86_64.sh -O /tmp/miniconda.sh && \
+    echo 'export PATH=$CONDA_DIR/bin:$PATH' > /etc/profile.d/conda.sh && \
+    /bin/bash /tmp/miniconda.sh -b -p $CONDA_DIR && \
+    rm -rf /tmp/* && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
 
-# The code to run when container is started:
-#COPY run.py .
-ENTRYPOINT ["conda", "run", "--no-capture-output", "-n", "geo_deep_env", "python"]
\ No newline at end of file
+## Variables 
+ARG GIT_HASH=f7e6033
```

Comment:
Since the type of tag you are referring to is "[lightweight](https://git-scm.com/book/en/v2/Git-Basics-Tagging)", it’s just a pointer to a specific commit hash.  For deployment purposes we certainly want to build images with a (version) tag.  For development purposes however I think it makes sense to build images on specific commits/hashes.  Ideally we would be able to supply build args of either `hash` or `tag` type.  

I asked a question a few weeks ago that is akin to what I'd like to be able to do here.  In [that case](https://stackoverflow.com/questions/70840481/corporate-ca-certificate-in-git-repo) it was about optionally including a corporate certificate at build time.  I have not found satisfactory answers yet.  I could modify the PR to accept tags instead of hashes but if we'll want to use the Dockerfile to build containers dynamically on committed pushes it does not make sense IMO.

Reply:
I understand you are looking at the maximum flexibility possible when cloning, but I still disagree with the use of hash. Git hashing is useful as an "internal unique identifier" for git, but I don't think it's meant to be used the way we use it here. I see two main disadvantages:
- Whether it's tags of hashes the user will still have to manually change the Dockerfile to get the version of GDL he/she wants (a "latest" tag like in Dockerhub could be cool). If using hashes, this means one needs to get the desired hash everytime one wants to perform a git clone. Since git hashes don't "mean anything" in terms of GDL's state (i.e. hashes don't give you a hint on the year/month, PR or anything else you are about to clone), it seems like pointing to a specific tag is more "meaningful". For the purpose of using the Dockerfile successfully, the presupposes that the developers of GDL push a tag **at least** when dependencies change (which happens no more than a few times a year). Otherwise, we could require devs to push a tag at every bugfix or so (2.0.1 -> 2.0.2).
- The other disadvantage is in the Docker build flow itself. By using a tag, and therefore making it possible to git clone with the [--depth ](https://www.git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt) parameter, the git history is automatically discarded. If using hash, the git history is downloaded (about 100 Mb), then deleted. This is not optimal and extends the build by a few unnecessary  seconds IMO.

Reply:
> Whether it's tags of hashes the user will still have to manually change the Dockerfile to get the version of GDL he/she want

No.  The HASH is a build argument : `--build-arg GIT_HASH=your-hash`.  We can edit the Dockerfile periodically to put a "good default" but other than that it is totally overridable at build time.  That way the Dockerfile is parametrized, so when doing CI for instance you could dynamically build an image with the latest commit for tests.  If only we could specify either HASH or TAG, that would make the Dockerfile very agile.

For now, I suggest you test the resulting image and make sure it works as expected.  If it does, then we'll have done "iteration 1" of Dockerfile optimization and I think we could merge this PR.  Note : The image now weighs 9.5GB for a reason I can't understand ...

About that sudden extra GB that came about with this last version of Dockerfile, I would open another issue with the idea of specializing Dockerfiles : one for CI (probably a one-stage who-cares-how-fat-it-is), one for publication (e.g. on DockerHub with git TAGS) and one for inference, as it does not require all the dependencies we pack in our Docker image.  How does that sound ?

Reply:
Ok for setting HASH as build argument, but still, 

> this means one needs to get the desired hash everytime one wants to perform a git clone

. The disadvantage I pointed out is still there.

I approved the PR for the sake of moving forwards beyond this debate, but I'm still far from convinced by the use git hash here.